heidrun/ufl_marc.rb

```diff
@@ -0,0 +1,108 @@
+
+def caribbean?(parser_value)
+  parser_value.value.include?('Digital Library of the Caribbean')
+end
+
+Krikri::Mapper.define(:ufl_marc, :parser => Krikri::MARCXMLParser) do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/ufl'
+    label 'University of Florida Libraries'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent,
+               :each => record.field('marc:datafield')
+                              .match_attribute(:tag, '535'),
+               :as => :dataP do
+    providedLabel dataP.field('marc:subfield').match_attribute(:code, 'a')
+  end
+
+  intermediateProvider :class => DPLA::MAP::Agent,
+                       :each => record.field('marc:datafield')
+                                      .match_attribute(:tag, '830')
+                                      .field('marc:subfield')
+                                      .match_attribute(:code, 'a')
+                                      .select { |a| caribbean?(a) },
+                       :as => :ip do
+    providedLabel ip
+  end
+  
+  isShownAt :class => DPLA::MAP::WebResource,
+            :each => record.field('marc:datafield')
+                           .match_attribute(:tag, '856'),
+            :as => :the_uri do
+    uri the_uri.field('marc:subfield').match_attribute(:code, 'u')
+  end
+
+  preview :class => DPLA::MAP::WebResource,
+          :each => record.field('marc:datafield')
+                         .match_attribute(:tag, '992'),
+          :as => :thumb do
+    uri thumb.field('marc:subfield').match_attribute(:code, 'a')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+
+    collection :class => DPLA::MAP::Collection, 
+               :each => record.field('marc:datafield')
+                              .match_attribute(:tag, '830'),
+               :as => :coll do
+      title coll.field('marc:subfield').match_attribute(:code, 'a')
+    end
+
+    #don't know how to do these but am keeping placeholder
+    #contributor 
+
+    #creator 
```

Comment:
The creator mapping suggests that we need only `$a` for 100, 110, 111 (and `700` when `$e` is `joint author` or `jt author`). Do we really want to throw away the other subfields?